After removing a source from a source stream the NATS Server crashes

## Defect
When I remove a source from a source stream it causes a fatal crash of the NATS Server a few seconds after the stream has been updated.  Also when I list streams right after the removal of a source, the stream still is showing a source but it is an incomplete fragment of the original source. This can all be seen by debugging the code in the linked gist.

Make sure that these boxes are checked before submitting your issue -- thank you!

 - [x ] Included `nats-server -DV` output
 [nats-server dv.txt](https://github.com/nats-io/nats-server/files/7590219/nats-server.dv.txt)
 - [ x] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)
 https://gist.github.com/kylebernhardy/a3d64e93175ec92bc3859fe2cc522bf0

#### Versions of `nats-server` and affected client libraries used:
nats-server 2.6.5
nats.js 2.4.0

#### OS/Container environment:
Ubuntu 20.04

#### Steps or code to reproduce the issue:
Run the commands `nats-server -c ./hub.json` `nas-server -c ./leaf.json` (these server configs are included in the linked gist)
Run/debug the test.js script

#### Expected result:
Desired source is successfully removed from stream & server does not crash.

#### Actual result:
The stream still has a partial source fragment & the server has a fatal crash.

you can view my simple topology in the gist with the server configs. Here is the panic output:

29220] 2021/11/22 22:06:04.822146 [DBG] Retrying source consumer for 'HDB > hdb_work_queue'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x985349]

goroutine 127 [running]:
github.com/nats-io/nats-server/server.(*stream).setSourceConsumer(0xc000232000, 0xc000138780, 0x10, 0x2)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/stream.go:1792 +0x149
github.com/nats-io/nats-server/server.(*stream).retrySourceConsumerAtSeq(0xc000232000, 0xc000138780, 0x10, 0x2)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/stream.go:1755 +0x185
github.com/nats-io/nats-server/server.(*stream).retrySourceConsumer(0xc000232000, 0xc000138780, 0x10)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/stream.go:1739 +0xf0
github.com/nats-io/nats-server/server.(*stream).processSourceMsgs(0xc000232000, 0xc0006121e0)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/stream.go:1961 +0x32c
github.com/nats-io/nats-server/server.(*stream).setSourceConsumer.func1()
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/stream.go:1803 +0x33
created by github.com/nats-io/nats-server/server.(*Server).startGoRoutine
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/server.go:2867 +0xc5

--- a/server/stream.go
+++ b/server/stream.go
@@ -1787,6 +1787,9 @@ func (mset *stream) setSourceConsumer(iname string, seq uint64) {
 	si.sseq, si.dseq = seq, 0
 	si.last = time.Now()
 	ssi := mset.streamSource(iname)
+	if ssi == nil {
+		return
+	}
 
 	// Determine subjects etc.
 	var deliverSubject string

